### PR TITLE
Add migration to fix user verification state

### LINKED ISSUE #375: Users appear as verified when they aren't
### Current behavior
Some users have reported not receiving the balance-based rewards even though they were verified and had the minimum required balance. However, upon closed inspection, they show up as verified in the app but have only 2 successful attestations.

### Desired behavior
Users never show up as verified unless they have 3 completed attestations.

### Steps to Reproduce
Undetermined. The hypothesis is that this is due to a bug on 1.13

### Versions

This issue started after the 1.13 update that had other verification-related issues.

### LINKED ISSUE #376: Migrate the `mtwAddress` from `unverifiedMtwAddress` if needed.
### Current behavior
In 1.13 some users verified but we had a bug which caused us not to call `setMtwAddress` once verification succeeds, so some users verified successfully but don't have their `mtwAddress` stored in the Redux store.

### Desired behavior
They do have a `unverifiedMtwAddress`, so we can add a migration of sorts that checks if that address is verified or not and calls `setMtwAddress` if it is.

### Steps to Reproduce
Verify your account on 1.13. You will not be able to redeem escrow payments since your `mtwAddress` is not set

### Versions

This was caused by a bad deploy on 1.13

### PR DESCRIPTION
**Pull request by gnardini**
### Description

Added a migration that fixes two errors introduced by the 1.13 release:
- `mtwAddress` not set in redux store
- Non-verified users are marked as verified

### Other changes

N/A

### Tested

Manually by setting the state of the redux state directly.
Unit tests

### How others should test

- Use the 1.13 build to verify while using repeated codes.
- See that you show up as verified.
- Update the app to 1.14.2.
- See that you no longer show up as verified.

### Related issues

- Fixes #375
- Fixes #376

### Backwards compatibility

N/A

### CODE REVIEW

File: packages/mobile/src/app/reducers.ts [hunk 1]
```diff
@@ -23,6 +23,9 @@ export interface State {
   hideVerification: boolean
   showRaiseDailyLimitTarget: string | undefined
   walletConnectEnabled: boolean
+  // In 1.13 we had a critical error which requires a migration to fix. See |verificationMigration.ts|
+  // for the migration code. We can remove all the code associated with this after some time has passed.
+  ranVerificationMigration: boolean
```
Comment (jeanregisser) [inline packages/mobile/src/app/reducers.ts:28]: What do you think of using a timestamp instead?
https://changelog.com/posts/you-might-as-well-timestamp-it
```suggestion
  ranVerificationMigrationAt: number | undefined 
```

**Review by jeanregisser — COMMENTED** [review]
Looking great!

File: packages/mobile/src/app/verificationMigration.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  const address = mtwAddress ?? komenci.unverifiedMtwAddress
+  Logger.debug(
+    TAG,
+    `mtwAddress: ${mtwAddress} - unverifiedMtwAddress: ${komenci.unverifiedMtwAddress}`
+  )
+
+  const isVerified: boolean = yield call(isAddressVerified, address)
+  Logger.debug(TAG, `address ${address} is verified: ${isVerified}`)
+  yield put(verificationMigrationRan(isVerified ? address : null, isVerified))
+}
```
[the remark this answers is not in the document]
Comment (gnardini) [reply packages/mobile/src/app/verificationMigration.ts]: Skipping changing everything if there's no `mtwAddress` nor `unverifiedMtwAddress` in the store :)

**Review by jeanregisser — APPROVED** [review]
🙏 👍

**Pull request merged**